### ISSUE TITLE
Bash completion uses unknown command `have`

### ISSUE THREAD

**Issue by shepmaster** [rust-lang/cargo#1509]
[This code](https://github.com/rust-lang/cargo/blob/72c2e3d3807995c624d1010e427e3ca5d14eb84c/src/etc/cargo.bashcomp.sh#L1):

```bash
have cargo &&
```

Uses `have`, which neither my OS X or Linux box recognize as a command.

A quick perusal of the [bash builtins](http://www.gnu.org/software/bash/manual/html_node/Bash-Builtins.html) list didn't seem to include the `have` command, either. For the last 3-5 months, I've just been editing this file and removing that line to get completion working. I must be missing some part of the infrastructure that provides this. What documentation am I failing to read? :smile_cat:

**Comment by Potpourri** [comment]
`have` is bash function-helper, it defined in your system `bash_completion` (on my linux it `/usr/share/bash-completion`). `have` is available only for completions loaded by `bash_completion`.
Tl;dr add `. /path/to/bash_completion.d/cargo` to `~/.bash_completion`

**Comment by shepmaster** [comment]
@Potpourri Thanks! I guess the part I was missing is that the completion provided by Cargo is intended to work with bash_completion, not just to be sourced by bash directly. Is this something that should be documented (or is it already somewhere)? Perhaps this is just one of those things that everyone already knows about?

**Comment by Potpourri** [comment]
This is the standard way to connect completions provided by bash. Probably described in the bash documentation.

**Comment by shepmaster** [comment]
OK, then I'll chalk this up as a learning exercise for me. Thanks!
